driver/modern_bpf/programs/tail_called/events/syscall_dispatched_events/quotactl.bpf.c

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright (C) 2022 The Falco Authors.
+ *
+ * This file is dual licensed under either the MIT or GPL 2. See MIT.txt
+ * or GPL2.txt for full copies of the license.
+ */
+
+#include <helpers/interfaces/fixed_size_event.h>
+#include <helpers/interfaces/variable_size_event.h>
+
+/*=============================== ENTER EVENT ===========================*/
+
+SEC("tp_btf/sys_enter")
+int BPF_PROG(quotactl_e,
+	     struct pt_regs *regs,
+	     long syscall_id)
+{
+	struct ringbuf_struct ringbuf;
+	if(!ringbuf__reserve_space(&ringbuf, QUOTACTL_E_SIZE))
+	{
+		return 0;
+	}
+
+	ringbuf__store_event_header(&ringbuf, PPME_SYSCALL_QUOTACTL_E, QUOTACTL_E_SIZE);
+
+	/*=============================== COLLECT PARAMETERS  ===========================*/
+
+	/* Parameter 1: cmd (type: PT_FLAGS16) */
+	unsigned long cmd = extract__syscall_argument(regs, 0);
```

Comment:
Hi @Andreagit97 ,

here I get an test failure on s390x:
````
[ RUN      ] SyscallEnter.quotactlE
/root/git/falcosecurity/libs/test/modern_bpf/event_class/event_class.cpp:212: Failure
Expected equality of these values:
  *(T*)(m_event_params[m_current_param].valptr)
    Which is: 0
  param
    Which is: 128
>>>>> value of the param is not correct. Param id = 1

[  FAILED  ] SyscallEnter.quotactlE (0 ms)
````

With some `bpf_printk()` instrumentation:
````c
--- a/driver/modern_bpf/programs/tail_called/events/syscall_dispatched_events/quotactl.bpf.c
+++ b/driver/modern_bpf/programs/tail_called/events/syscall_dispatched_events/quotactl.bpf.c
@@ -27,6 +27,7 @@ int BPF_PROG(quotactl_e,
 
        /* Parameter 1: cmd (type: PT_FLAGS16) */
        unsigned long cmd = extract__syscall_argument(regs, 0);
+       bpf_printk("quotctl cmd=%x (%x) scap=%x\n", cmd, cmd >> SUBCMDSHIFT, quotactl_cmd_to_scap(cmd));
        u16 scap_cmd = quotactl_cmd_to_scap(cmd);
        ringbuf__store_u16(&ringbuf, scap_cmd);
 
````
the following is printed:
````
        bpf_test-11439   [001] d.... 326644.488183: bpf_trace_printk: quotctl cmd=80000100 (ff800001) scap=0
````

Changing from `unsigned long` (s390x defaults to 64-bit) to `unsigned int` (s390x defaults to 32-bit) let the test run successful:
````c
--- a/driver/modern_bpf/programs/tail_called/events/syscall_dispatched_events/quotactl.bpf.c
+++ b/driver/modern_bpf/programs/tail_called/events/syscall_dispatched_events/quotactl.bpf.c
@@ -26,7 +26,8 @@ int BPF_PROG(quotactl_e,
        /*=============================== COLLECT PARAMETERS  ===========================*/
 
        /* Parameter 1: cmd (type: PT_FLAGS16) */
-       unsigned long cmd = extract__syscall_argument(regs, 0);
+       unsigned int cmd = extract__syscall_argument(regs, 0);
+       bpf_printk("quotctl cmd=%x (%x) scap=%x\n", cmd, cmd >> SUBCMDSHIFT, quotactl_cmd_to_scap(cmd));
        u16 scap_cmd = quotactl_cmd_to_scap(cmd);
        ringbuf__store_u16(&ringbuf, scap_cmd);
 
````
and for completeness the trace output:
````
bpf_test-11557   [000] d.... 326958.038886: bpf_trace_printk: quotctl cmd=80000100 (800001) scap=80
````

For reference, the Linux kernel quota syscall also uses an [unsigned int](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/fs/quota/quota.c#n916).

Reply:
I will also give #550 a test run before I leave for vacation.

Reply:
Thank you very much for this @hbrueckner! The fix seems good I will try it!

Reply:
I've applied the solution!! Thank you for the fix as always :heart: